File: src/ServerScriptService/GroupChecker.server.lua
local BRAINBLOX_ID = 15553074
local MINIMUM_RANK = 1

local function OnPlayerAdded(player: Player)
	if player:GetRankInGroup(BRAINBLOX_ID) < MINIMUM_RANK then
		game.ReplicatedStorage.Kicked:FireClient(player)
		player:Kick("not in group")
	end
end

if not game:GetService("RunService"):IsStudio() then 
	for _,player: Player in game.Players:GetPlayers() do
		OnPlayerAdded(player)
	end
	game.Players.PlayerAdded:Connect(OnPlayerAdded)
end

File: src/server/scripts/group-checker.server.ts
import { Players, RunService } from "@rbxts/services";
import { Events } from "server/modules/networking";

const BRAINBLOX_ID = 15553074;
const MINIMUM_RANK = 1;

function onPlayerAdded(player: Player): void {
	if (player.GetRankInGroup(BRAINBLOX_ID) >= MINIMUM_RANK) return;
	Events.kicked(player);
	player.Kick("not in group");
}

if (!RunService.IsStudio()) {
	Players.GetPlayers().forEach((player) => onPlayerAdded(player));
	Players.PlayerAdded.Connect(onPlayerAdded);
}
